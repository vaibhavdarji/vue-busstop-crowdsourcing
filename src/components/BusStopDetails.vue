<template>
  <v-container fluid grid-list-md>
    <template v-if="errorMsg">
      <custom-error :msg="errorMsg" @onErrorBtnClick="onRefresh"></custom-error>
    </template>
    <template v-if="currentBusStop">
      <v-layout row wrap justify-center>
        <v-flex xs8>
          <stepper></stepper>
        </v-flex>
      </v-layout>
    </template>
    <template v-if="!errorMsg && !currentBusStop">
      <v-layout row wrap justify-center>
        <v-flex xs6 align-center>
          <v-select
            :items="totalStops"
            label="Select bus stop to donate"
            item-text="name"
            item-value="name"
            persistent-hint
            return-object
            single-line
            @change="onChangeBusSelect"
          ></v-select>
        </v-flex>
      </v-layout>
    </template>
  </v-container>
</template>



<script>
import { mapGetters, mapActions } from 'vuex';
import Stepper from '@/components/Stepper.vue';
import CustomError from '@/components/Error.vue';
import {
  GET_ALL_STOPS,
  SET_SELECTED_STOP
} from '@/actions/types';

export default {
  name: 'BusStopDetails',
  data () {
    return {
      errorMsg: '',
      doRefresh: false
    };
  },

  components: {
    Stepper,
    CustomError
  },

  asyncComputed: {
   totalStops: {
     get () {
       return this.busStops.length ? this.busStops: this.getAllStops().then((data) => {
         this.doRefresh = false;
         return data;
       }).catch((error) => {
         this.errorMsg = error.message;
       });
     },

     default() {
       return [];
     },
     watch () {
      this.doRefresh
      }
   },
  },

  computed: {
    ...mapGetters(['selectedBusStop', 'busStops']),
    currentBusStop () {
      return this.selectedBusStop;
    }
  },

  methods: {
    ...mapActions({
      'getAllStops': GET_ALL_STOPS,
      'setSelectedStop': SET_SELECTED_STOP
    }),
    onChangeBusSelect(stop) {
      this.setSelectedStop(stop);
    },
    onRefresh () {
      this.errorMsg = '';
      this.doRefresh = true;
    }
  }
}
</script>

