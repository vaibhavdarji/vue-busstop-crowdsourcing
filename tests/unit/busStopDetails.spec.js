import { shallowMount } from "@vue/test-utils";
import BusStopDetails from '@/components/BusStopDetails.vue';
import Vue from "vue";
import Vuex from "vuex";
import Vuetify from "vuetify";

Vue.use(Vuex);
Vue.use(Vuetify);

const dummyData = {stopId: 1, lat: 33.760262, lng: -84.384706, donationsRaisedInDollars: 0, name: 'Hertz at Portman Blvd'};

describe("BusStopDetails.vue | with selected bus stop", () => {
  let getters, store;
  beforeEach(() => {
    getters = {
      selectedBusStop: () => dummyData
    };
    store = new Vuex.Store({
      getters
    });
  });

  it("should match snapshot", () => {
    const wrapper = shallowMount(BusStopDetails, {
      store,
      Vue
    });

    wrapper.setData({
      errorMsg: '',
      doRefresh: false
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should validate selected bus stop", () => {
    const wrapper = shallowMount(BusStopDetails, {
      store,
      Vue
    });

    wrapper.setData({
      errorMsg: '',
      doRefresh: false
    });
    expect(wrapper.vm.currentBusStop).toEqual(getters.selectedBusStop());
  });
});
