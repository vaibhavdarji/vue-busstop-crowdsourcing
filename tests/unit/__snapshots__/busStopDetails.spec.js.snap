// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusStopDetails.vue | with selected bus stop should match snapshot 1`] = `
<v-container-stub
  fluid=""
  grid-list-md=""
  tag="div"
>
  <!---->
   
  <v-layout-stub
    justify-center=""
    row=""
    tag="div"
    wrap=""
  >
    <v-flex-stub
      tag="div"
      xs8=""
    >
      <stepper-stub />
    </v-flex-stub>
  </v-layout-stub>
   
  <!---->
</v-container-stub>
`;
